packages/frontend/src/hooks/useApi.ts

```diff
@@ -1,20 +1,140 @@
 import useSWR from 'swr';
+import { paths } from '../types/api-schema';
 
-interface UseApiArgs {
+interface UseApiArgs<
+  TUrl extends keyof paths,
```

Comment:
should we consider to have names with more clarity? For instance, TUrl as TempUrl. I was a little confused about the meaning of T at first.

Reply:
The T stands for Type, these params are type parameters

Reply:
Its my preferred notation for TypeParameters rather than the typical single letter <K> <T> etc normally seen.